### ISSUE TITLE
Find widget: screen reader double speaks checkboxes title

### ISSUE THREAD

**Issue by jhomme** [microsoft/vscode#96077]
Issue Type: <b>Bug</b>

To find a few examples:

1. Press CTRL+F or CTRL+H and tab through those dialogs with NVDA.
2. Notice that the field prompts for many fields speak twice.
3. Check the NVDA log with NVDA+F1 and note that the Description property is the same as the Name property. The Description is the HTML title attribute.
4. You can check with JAWS, use INSERT+SHIFT+F1 and read the attributes for the current tag and note that the Title attribute is filled in with the same content as the Label attribute on some fields, such as edit boxes and list fields. JAWS covers up the accessibility error by eliminating the Title attribute from the verbal output, however. 

VS Code version: Code - Insiders 1.45.0-insider (d342048538c28d54afd1c6beea03f3a89af04359, 2020-04-23T05:57:41.076Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz (4 x 2397)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.87GB (3.58GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
hugofy|akm|0.1.1
markdown-toc|Ala|1.5.6
vscode-markdownlint|Dav|0.35.0
vscode-pandoc|Dou|0.0.8
markdown-shortcuts|mdi|0.12.0
python|ms-|2020.4.74986

</details>
We have written the needed data into your clipboard because it was too large to send. Please paste.

**Comment by isidorn** [comment]
Thanks for reproting this. Yes I can reproduce this with NVDA, however it works properly with Orca and VoiceOver.
The issue is that NVDA and JAWS read both the aria-label and the title. Or the description.
Thus I believe this is a bug in NVDA, that NVDA should check if the aria-label and title are the same it should not re-read the spoken text.

We already enounceterd something like this and the only thing that VS Code could do is explicitly set `aria-description=''` to override the title value. I tried this and this does not work with NVDA since it is probably only supported by orca atm.

@leonardder let me know if you have some suggestions or is it fair to say this is an NVDA issue.
Since these issues seem to capture exactly that
https://github.com/nvaccess/nvda/issues/10548
https://github.com/nvaccess/nvda/issues/7841

**Comment by jhomme** [comment]
Hi,

Hopefully an NVDA expert will see this. I believe NVDA does not omit any accessibility info, but I could be totally wrong. It may be necessary to enhance whatever add-on may exist for VS Code or make one to help with this. Personally, and this is just my opinion as someone who works in accessibility, if ARIA labels say the same thing as field labels, the ARIA labels should disappear as long as the screen reader sees and speaks the field label, because that is the more semantic way to do things, and is universal.

If I am misunderstanding the issue, please help me understand.

 

Thanks.

 

Jim H

 

==========

Jim Homme

Skype: jim.homme

FreeChess: jhomme

Twitter: jimhomme

Facebook: jimhomme

**Comment by isidorn** [comment]
Also fyi @marcozehe in case you know someone over at nvda to ping apart from @leonardder who I ping all the time :)

**Comment by MarcoZehe** [comment]
I added my endorsement to the NVDA issues. Firefox does strip AccDescription if it turns out to be identical to AccName, but that's a choice we made 10 years ago or so. And it proves that this is a practical approach since both JAWS and Orca do it, too. I am all for NVDA doing this as well.
